File: Lista-15/java/4.java
class Animal {
    public String som() {
        return "";
    }
}

class Cachorro extends Animal {
    @Override
    public String som() {
        return "Latido";
    }
}

class Gato extends Animal {
    @Override
    public String som() {
        return "Miau";
    }
}

public class Main {
    public static void main(String[] args) {
        Cachorro cachorro = new Cachorro();
        Gato gato = new Gato();
        System.out.println(cachorro.som());
        System.out.println(gato.som());
    }
}

File: Lista-15/go/4.go
package main
import "fmt"

type Animal interface {
    Som() string
}

type Cachorro struct{}

func (c Cachorro) Som() string {
    return "Latido"
}

type Gato struct{}

func (g Gato) Som() string {
    return "Miau"
}

func main() {
    cachorro := Cachorro{}
    gato := Gato{}
    fmt.Println(cachorro.Som())
    fmt.Println(gato.Som())
}
